x/ibc-rate-limit/rate_limit.go

```diff
@@ -151,17 +152,19 @@ func GetLocalDenom(denom string) string {
 	}
 }
 
-func CalculateChannelValue(ctx sdk.Context, denom string, port, channel string, bankKeeper bankkeeper.Keeper) sdk.Int {
+func CalculateChannelValue(ctx sdk.Context, denom string, bankKeeper bankkeeper.Keeper, channelKeeper channelkeeper.Keeper) sdk.Int {
+	// For non-native (ibc) tokens, return the supply if the token in osmosis
 	if strings.HasPrefix(denom, "ibc/") {
 		return bankKeeper.GetSupplyWithOffset(ctx, denom).Amount
 	}
 
-	if channel == "any" {
-		// ToDo: Get all channels and sum the escrow addr value over all the channels
-		escrowAddress := transfertypes.GetEscrowAddress(port, channel)
-		return bankKeeper.GetBalance(ctx, escrowAddress, denom).Amount
-	} else {
-		escrowAddress := transfertypes.GetEscrowAddress(port, channel)
-		return bankKeeper.GetBalance(ctx, escrowAddress, denom).Amount
+	// For native tokens, obtain the balance held in escrow for all potential channels
+	channels := channelKeeper.GetAllChannels(ctx)
+	balance := sdk.NewInt(0)
+	for _, channel := range channels {
+		escrowAddress := transfertypes.GetEscrowAddress("transfer", channel.ChannelId)
+		balance = balance.Add(bankKeeper.GetBalance(ctx, escrowAddress, denom).Amount)
+
 	}
```

Comment:
I'm worried about this being too slow for native assets right now.

Do we want to comment out for native assets until we can get this into a pre-computed state entry?

Reply:
yeah, I was wondering about that too. 

We need the value to be calculated across all channels if we want to be able to filter the tokens for any channel (like we discussed). I can add the pre-computed state and a query for it, but it's a detour. 

We could also do the queries concurrently, but we still need to do one query per channel.

Reply:
Maybe we do native assets = bank supply, non-native = escrow account for now?

And then once we get an efficient state entry for the sum of IBC'd out native assets across all channels, we switch the go side?

Reply:
Made this issue in IBC-go to track this: https://github.com/cosmos/ibc-go/issues/2664

Reply:
That makes sense to me. I'll make the changes.

It's pretty much what we had before 😅, but should be easier to modify now since the tests are more flexible.

Reply:
sorry :(

Reply:
haha, no worries. Now we know what to do :)